src/Plugin/ZynAddSubFX/CMakeLists.txt

```diff
@@ -177,52 +89,30 @@ else()
     endif()
 endif()
 
-if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
-    target_link_libraries(ZynAddSubFX_lv2 zynaddsubfx_core ${OS_LIBRARIES} ${LIBLO_LIBRARIES}
-        ${PLATFORM_LIBRARIES})
-endif()
-if(APPLE)
-    target_link_libraries(ZynAddSubFX_vst zynaddsubfx_core ${OS_LIBRARIES} ${LIBLO_LIBRARIES}
-        ${PLATFORM_LIBRARIES} "-framework Cocoa" "-framework openGL" "-Wl,-exported_symbol,_VSTPluginMain")
-else()
-    target_link_libraries(ZynAddSubFX_vst zynaddsubfx_core ${OS_LIBRARIES} ${LIBLO_LIBRARIES}
-        ${PLATFORM_LIBRARIES})
-endif()
-if(ZestGui AND NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
-    if(APPLE)
-        target_link_libraries(ZynAddSubFX_lv2_ui "-framework Cocoa" "-framework openGL" "-Wl,-exported_symbol,_lv2ui_descriptor")
-        target_link_libraries(ZynAddSubFX_lv2 "-Wl,-exported_symbol,_lv2_descriptor" "-Wl,-exported_symbol,_lv2_generate_ttl")
-    else()
-        target_link_libraries(ZynAddSubFX_lv2_ui X11 GL)
-    endif()
-endif()
-
-if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
-    install(TARGETS ZynAddSubFX_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynAddSubFX.lv2/)
-    install(TARGETS ZynAddSubFX_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
+target_link_libraries(ZynAddSubFX-dsp PRIVATE zynaddsubfx_core zynaddsubfx_gui_stubs
+    ${OS_LIBRARIES} ${LIBLO_LIBRARIES} ${PLATFORM_LIBRARIES})
 
-    add_custom_command(TARGET ZynAddSubFX_lv2 POST_BUILD
-        COMMAND ../../lv2-ttl-generator $<TARGET_FILE:ZynAddSubFX_lv2>
-        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lv2)
+# Linker diagnostic options
+target_link_libraries(ZynAddSubFX PRIVATE -Wl,--no-undefined,--unresolved-symbols=report-all)
```

Comment:
this is not compatible with apple clang, but we can sort this after merging as the dpf-ci-workflow PR has build tests for macOS which will tell us any such build issues

Reply:
This is used for diagnostics when I did the porting stuff.

If ZynAddSubFX can build and run well with Apple Clang, I can remove this.